src/test/test-path.c

```diff
@@ -100,17 +100,23 @@ static int _check_states(unsigned line,
                          service_state_to_string(service->state),
                          service_result_to_string(service->result));
 
-                if (service->state == SERVICE_FAILED &&
-                    service->main_exec_status.status == EXIT_CGROUP &&
-                    !ci_environment())
+                if (service->state == SERVICE_FAILED && service->main_exec_status.status == EXIT_CGROUP) {
+                        const char *ci = ci_environment();
+
                         /* On a general purpose system we may fail to start the service for reasons which are
                          * not under our control: permission limits, resource exhaustion, etc. Let's skip the
                          * test in those cases. On developer machines we require proper setup. */
-                        return log_notice_errno(SYNTHETIC_ERRNO(ECANCELED),
-                                                "Failed to start service %s, aborting test: %s/%s",
-                                                UNIT(service)->id,
-                                                service_state_to_string(service->state),
-                                                service_result_to_string(service->result));
+                        if (!ci)
+                                return log_notice_errno(SYNTHETIC_ERRNO(ECANCELED),
+                                                        "Failed to start service %s, aborting test: %s/%s",
+                                                        UNIT(service)->id,
+                                                        service_state_to_string(service->state),
+                                                        service_result_to_string(service->result));
+
+                        /* On Salsa we can't setup cgroups so the unit always fails */
+                        if (streq(ci, "salsa-ci"))
+                                exit(EXIT_TEST_SKIP);
```

Comment:
> as we cannot set up cgroups 

If so, how about to check if we can set up cgroups at the beginning of the test?

Reply:
is there a helper for that?

Reply:
Not sure it is directory applied for this test, but, in several tests, we set up cgroup sub tree and skip if we cannot.
E.g. 46f0a4e7ac97b47c5718b718e3c6db4a8a8e78a0

Reply:
this test already does that but continues if it cannot, explicitly, because in some conditions it can still work. But not on this particular CI environment, hence, the specific skip.

Reply:
OK. This is just a test code. Let's take the currently proposed way until we find a better way.

Reply:
But, could you extend the comment about that?